pkgs/development/compilers/javacard-devkit/default.nix

```diff
@@ -0,0 +1,86 @@
+{ stdenv, requireFile, unzip, makeWrapper, oraclejdk8 }:
+
+let
+  pkg = "javacard-devkit";
+  version = "2.2.2";
+
+  underscoreVersion = builtins.replaceStrings ["."] ["_"] version;
+
+  downloadUrl = "http://www.oracle.com/technetwork/java/javasebusiness/downloads/java-archive-downloads-javame-419430.html#java_card_kit-2.2.2-oth-JPR";
+in
+
+stdenv.mkDerivation rec {
+  name = "${pkg}-${version}";
+
+  src = requireFile {
+    name = "java_card_kit-${underscoreVersion}-linux.zip";
+    url = downloadUrl;
+    sha256 = "1rzkw8izqq73ifvyp937wnjjc40a40drc4zsm0l1s6jyv3d7agb2";
+  };
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  unpackPhase = "true";
+
+  buildPhase = ''
+    unzip $src
+
+    cd "java_card_kit-${underscoreVersion}"
+    unzip "java_card_kit-${underscoreVersion}-rr-bin-linux-do.zip"
+
+    chmod +x bin/{apdutool,capdump,capgen,converter,cref,exp2text,jcwde,scriptgen,verifycap,verifyexp,verifyrev}
+    chmod u+w bin/cref
+    patchelf --set-interpreter "$(cat "$NIX_CC/nix-support/dynamic-linker")" bin/cref
```

Comment:
There is also `libjpcsclite.so` (not sure which component uses it however), which needs `libpcsclite.so`. It might be a good idea to use `autoPatchelfHook` here.

Reply:
I don't know where that's used either…

Good catch for noticing it needs `libpcsclite.so`! it also appears to need `libnsl.so.1` and `libc.so.6`. However, for some reason I can't understand, adding `glibc` to the `buildInputs` doesn't appear to make `autoPatchelfHook` auto-correct it. I've now tried quite a number of variations of it, so if you have any idea about what's going wrong…

(the fact that `libnsl-1.1.0` appears to package `libnsl.so.2.0.0` and the fact I'm not even sure it is used somewhere are probably just as important reasons for my decision to ragequit this issue after a few hours, though :°)

Reply:
Adding `glibc` to `buildInputs` shouldn't be needed. What's the output from `ldd`?

Reply:
There's no output from `ldd` (it's a shared library), but here is the output from `readelf -d`:
```
Dynamic section at offset 0x20a0 contains 23 entries:
  Tag        Type                         Name/Value
 0x0000001d (RUNPATH)                    Library runpath: [/nix/store/n962knaz7h3qjdjf01g6r0dwld9qswyc-pcsclite-1.8.23/lib]
 0x00000001 (NEEDED)                     Shared library: [libpcsclite.so]
 0x00000001 (NEEDED)                     Shared library: [libc.so.6]
 0x00000001 (NEEDED)                     Shared library: [libnsl.so.1]
 0x0000000c (INIT)                       0x50c
 0x0000000d (FINI)                       0x684
 0x00000004 (HASH)                       0xb4
 0x00000005 (STRTAB)                     0x2188
 0x00000006 (SYMTAB)                     0x18c
 0x0000000a (STRSZ)                      240 (bytes)
 0x0000000b (SYMENT)                     16 (bytes)
 0x00000003 (PLTGOT)                     0x17a0
 0x00000002 (PLTRELSZ)                   16 (bytes)
 0x00000014 (PLTREL)                     REL
 0x00000017 (JMPREL)                     0x4fc
 0x00000011 (REL)                        0x4d4
 0x00000012 (RELSZ)                      40 (bytes)
 0x00000013 (RELENT)                     8 (bytes)
 0x6ffffffe (VERNEED)                    0x4b4
 0x6fffffff (VERNEEDNUM)                 1
 0x6ffffff0 (VERSYM)                     0x46c
 0x6ffffffa (RELCOUNT)                   2
 0x00000000 (NULL)                       0x0
```

So from what I can tell it's not adding `${glibc}/lib` (for `libc.so.6`) in the `RUNPATH`. (and it didn't add the `libnsl.so.1` either but that's pretty normal) Hence my attempts to add it into `buildInputs` :)

Reply:
After discussion over IRC, I was just being stupid and assuming that `nix-shell --pure -p pkgsi686Linux.glibc` actually was pure, while it actually gave the 64-bits `ldd` (due to the internals of `-p`).

So with the actual `ldd`, that does perform recursive resolution of the runpath, the problem is solved and both `libc.so.6` and `libnsl.so.1` are already provided by the runpath. So no longer any problem here :)